docs/Compile.rst

```diff
@@ -697,55 +704,89 @@ during a build.  If you get a failure, you may miss important errors or warnings
 
-* When you've installed Cygwin and its SSH server:
+Next, with Visual Studio open, check the build configuration drop downs. Debug does not work, so you must
+select ``RelWithDebInfo`` if you wish for debug information to be generated. Otherwise you can select
+``Release`` if you wish.
 
-  * You can now SSH in to your Windows install and compile from a remote terminal;
-    very useful if your Windows installation is a local VM on a \*nix host OS.
+Now you're ready to build. You'll find the Cmake defined targed under: ``CMakePredefinedTargets``
+You'll likely want to select ``INSTALL`` or ``ALL_BUILD`` from this folder.
 
-* In general: you can use Bash as your compilation terminal, meaning you have a decent
-  sized window, scrollback, etc.
+.. note::
 
-  * Whether you're accessing it locally as with Git's Bash, or remotely through
-    Cygwin's SSH server, this is far superior to using ``cmd.exe``.
+    If you're having trouble building, remember Visual Studio sometimes needs you
+    to rebuild or clean targets. So give that a try if you're seeing build errors.
+    Otherwise check the `FAQ`_ or `discord`_
 
-You don't need to do anything special to compile from Bash. As long as your PATHs
-are set up correctly, you can run the same generate- and build/install/package- bat
-files as detailed above.
+With the documentation
+~~~~~~~~~~~~~~~~~~~~~~
+
+To generate documentation, you'll need to enable it when you generate your build/project files.
+
+* When initially running CMake, add ``-DBUILD_DOCS:bool=ON`` to your ``cmake``
+  command. For example::
+
+    cmake .. -DCMAKE_BUILD_TYPE:string=Release -DBUILD_DOCS:bool=ON -DCMAKE_INSTALL_PREFIX=<path to DF>
+
+* If you have already run CMake, you can simply run it again from your build
+  folder to update your configuration::
+
+    cmake .. -DBUILD_DOCS:bool=ON
+
+* You can edit the ``BUILD_DOCS`` setting in CMakeCache.txt directly
+
+* You can use the CMake GUI or ``ccmake`` to change the ``BUILD_DOCS`` setting
+
+* On Windows, if you prefer to use the batch scripts, you can run
+  ``generate-msvc-gui.bat`` and set ``BUILD_DOCS`` through the GUI. If you are
+  running another file, such as ``generate-msvc-all.bat``, you will need to edit
+  it to add the flag. You can also run ``cmake`` on the command line, similar to
+  other platforms.
+
+The generated documentation will be stored in ``docs/html`` in the root DFHack
+folder, and will be installed to ``hack/docs`` when you next install DFHack in a
+DF folder.
+
+FAQ
+---
+
+**What do I do when I'm getting the error** ``fatal error C1083: Cannot open include file: 'CoreProtocol.pb.h': No such file or directory``
+**or** ``..\zlib\lib\zlib.lib : warning LNK4272: library machine type 'X86' conflicts with target machine type 'x64'``?
+
+* Check the architecture CMake has configured: ``cat CMakeCache.txt | grep ARCH`` if it is 32, and you're building 64 [or vice versa] you'll need to regenerate the project from scratch.
```

Comment:
This should probably be replaced with a windows-centric command or at least to specify to use git-bash or some such to support the command.

Reply:
I suggest that we just tell people to search for DFHACK_BUILD_ARCH (i.e. the full option name) in the file and let them figure out a way that works for them. The instructions as-is could turn up multiple lines that are more difficult to interpret without context.